### ISSUE TITLE
Notifications for Linux desktop

### ISSUE THREAD

**Issue by brokenpip3** [borgbase/vorta#84]
Hi,

i really like vorta! works great. 
It's possible to add a features for a mail report for each backup? 

Thanks

**Comment by m3nu** [comment]
I think adding all the mail settings and libraries (SMTP, etc) to Vorta would be overkill. For macOS we already have notifications sent to the notification center. For Linux I don't know of a unified mechanism so far.

I do know that Linux sends many notifications to a local mailbox. So one option would be to regard those "local mails" as notifications in Linux.

In general I think that notifications for *failed* backups are more useful than the other way round. If you run Vorta on a server and want general monitoring on the client-side, filtering the logs would be another way. For server-side monitoring, one could look at the remote repo there, as we do for [BorgBase.com](https://www.borgbase.com) alerts.

In short, some more infos about your use case would be helpful. Then maybe we see a universal need for the feature or find a simpler solution. After all we want to do one thing well and not build another email client.

**Comment by brokenpip3** [comment]
Hi, thanks for you kind reply

> In general I think that notifications for _failed_ backups are more useful than the other way round.

yep, failed, success, general error etc

>If you run Vorta on a server and want general monitoring on the client-side, filtering the logs would be another way.

yep, maybe one great feature is settings option for "log path"?
So server-side i can put it on /var/log/vorta/something and parse with external tool

> In short, some more infos about your use case would be helpful. Then maybe we see a universal need for the feature or find a simpler solution. 

I don't have a personal case, i wrote only to give a general suggestion for vorta. I want just to know my backup job result.

>After all we want to do one thing well and not build another email client.

Yes i understand, but i don't think add smtplib module is so heavy :)
Maybe the best things is a general notification configuration? 

example:

settings --> notification ---> select you notification method
- system (notify-send on linux, mac??)
- logfile
- standard output
- send as script argument (this is the best option because people can use it for sending mail, pushbullett notification, slack etc; in my case i can use it for sending me telegram notification) 

As i said, it's simply nice-to-have, vorta just work great right now. :)

Sorry for bad english

**Comment by ThomasWaldmann** [comment]
not sure whether sending emails is appropriate for a desktop app...

but if you do it, there must be the option (and it better be the default option) to send email in any case.

if you only send email on a failed backup, first you email setup will fail, then your backup will fail, and nobody will notice. if you also send mails for successful backups, you positively know that the backup was ok and you will also likely notice when the email sending stops working for some reason.

**Comment by m3nu** [comment]
Also not too happy about emailing. That's more for servers, as I said. A server shouldn't run Vorta at all. It should run `borg` directly or [`borgmatic`](https://github.com/witten/borgmatic) to wrap it. The latter has a simple way to run commands before and after.

I'd rather find a standard way to send notifications on Linux too. Like the mentioned [libnotify](https://wiki.archlinux.org/index.php/Desktop_notifications#Libnotify). (same thing for [Ubuntu](http://manpages.ubuntu.com/manpages/cosmic/man1/notify-send.1.html))

So instead of emails, I'd rather do this:

- [ ] Add `libnotify` for Linux notifications
- [ ] Have option to globally enable/disable notifications. "Display notifications when background tasks fail."
- [ ] Additional sub-option to notify about successful backups as well: "Also notify about successful background tasks."

**Comment by brokenpip3** [comment]
> Also not too happy about emailing. That's more for servers, as I said. A server shouldn't run Vorta at all. It should run `borg` directly or [`borgmatic`](https://github.com/witten/borgmatic) to wrap it. The latter has a simple way to run commands (like emailing) before and after.

you convinced me, for a desktop app does not make sense the email, much better than the simple notification. (i have my simple borg bash script before found vorta, i'll try borgmatic instead on my server)

> So instead of emails, I'd rather do this:
> 
>     * [ ]  Add `libnotify` for Linux notifications
> 
>     * [ ]  Have option to globally enable/disable notifications. "Display notifications when background tasks fail."
> 
>     * [ ]  Additional sub-option to notify about successful backups as well: "Also notify about successful background tasks."

Very nice, ty!

i can try some PR later then

**Comment by m3nu** [comment]
Good we agree. For Borgmatic, I have an [Ansible role](https://galaxy.ansible.com/m3nu/ansible_role_borgbackup) and when using [BorgBase.com](https://www.borgbase.com/), it will give you a pre-filled template. Quite easy to set up really.

A PR for Linux notifications would be welcome. I only have Ubuntu set up in a VM and don't use it daily.

**Comment by m3nu** [comment]
This could be used for Linux notifications? https://notify2.readthedocs.io/en/latest/

**Comment by m3nu** [comment]
Are you working on this, @brokenpip3 or should I go ahead and just implement the above library?

**Comment by ThomasWaldmann** [comment]
I would really suggest NOT to add email sending to vorta.

email is just a big PITA to configure. and even if you did that right, the server might still reject your mails if it thinks they look strange or it doesn't like your IP or whatever and reject them or put them into spam/junk folder where nobody will read them.

**Comment by mygithubthrowaway** [comment]
@ThomasWaldmann we agree on desktop notification.

@m3nu I'm sorry, i want to implement this but I'm very busy at work these days (end of year sucks).
Maybe i can close this issue and insert a PR when i'm ready to test it, for sure not in 2018.

**Comment by bpereto** [comment]
Somebody working on it?
something to learn with desktop nitifications :)

I will do a PR:
* [ ]  Add `notify2` for Linux notifications
* [ ]  Have option to globally enable/disable notifications. "Display notifications when background tasks fail."
* [ ]  Additional sub-option to notify about successful backups as well: "Also notify about successful background tasks."

**Comment by brokenpip3** [comment]
Very nice! ty :+1: 
I definitely spy on your PR so i can learn by comparing my coding idea of this specific feature with someone else :)
